cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/MRJobClient.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright © 2015 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.gateway.handlers;
+
+import co.cask.cdap.common.exception.NotFoundException;
+import co.cask.cdap.proto.MRJobInfo;
+import co.cask.cdap.proto.MRTaskInfo;
+import com.google.common.collect.Lists;
+import com.google.common.collect.Maps;
+import com.google.inject.Inject;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.mapred.Counters;
+import org.apache.hadoop.mapred.JobClient;
+import org.apache.hadoop.mapred.JobStatus;
+import org.apache.hadoop.mapred.TaskReport;
+import org.apache.hadoop.mapreduce.TaskCounter;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Wrapper around Hadoop JobClient
+ */
+public class MRJobClient {
+  private final Configuration hConf;
+
+  @Inject
+  public MRJobClient(Configuration hConf) {
+    this.hConf = hConf;
+  }
+
+  /**
+   * @param runId for which information will be returned.
+   * @return a {@link MRJobInfo} containing information about a particular MapReduce program run.
+   * @throws IOException if there is failure to communicate through the JobClient.
+   * @throws NotFoundException if a Job with the given runId is not found.
+   */
+  public MRJobInfo getMRJobInfo(String runId) throws IOException, NotFoundException {
+    JobClient jobClient;
+    JobStatus[] jobs;
+    try {
+      jobClient = new JobClient(hConf);
+      jobs = jobClient.getAllJobs();
+    } catch (Exception e) {
+      throw new IOException(e);
```

Comment:
I think it should be the responsibility of whoever catches the exception (in this case, its the Handler).
Otherwise, it gets logged twice.